platform/darwin/src/MGLShapeSource.h

```diff
@@ -210,6 +210,31 @@ MGL_EXPORT
  */
 @property (nonatomic, copy, nullable) NSURL *URL;
 
+#pragma mark Querying the source's features
+
+/**
+ Returns an array of map features for this source, filtered by the given predicate.
+ 
+ Each object in the returned array represents a feature for the
+ current style and provides access to attributes specified by the relevant
+ <a href="https://www.mapbox.com/mapbox-gl-style-spec/#sources">tile sources</a>.
+ The returned array includes features specified in vector and GeoJSON tile
+ sources but does not include anything from raster, image, or video sources.
```

Comment:
Now that this method lives on MGLShapeSource, it’s unnecessary to point out that it won’t return features from non-shape sources. It’s also important to note that this method may return features that exist in the source but are not currently drawn by a style layer.